crates/sui-types/src/base_types.rs

```diff
@@ -1099,25 +1105,59 @@ impl TxContext {
         self.epoch
     }
 
-    /// Derive a globally unique object ID by hashing self.digest | self.ids_created
-    pub fn fresh_id(&mut self) -> ObjectID {
-        let id = ObjectID::derive_id(self.digest(), self.ids_created);
+    pub fn sender(&self) -> SuiAddress {
+        self.sender.into()
+    }
 
-        self.ids_created += 1;
-        id
+    pub fn epoch_timestamp_ms(&self) -> u64 {
+        self.epoch_timestamp_ms
     }
 
     /// Return the transaction digest, to include in new objects
     pub fn digest(&self) -> TransactionDigest {
         TransactionDigest::new(self.digest.clone().try_into().unwrap())
     }
 
-    pub fn sender(&self) -> SuiAddress {
-        SuiAddress::from(ObjectID(self.sender))
+    pub fn sponsor(&self) -> Option<SuiAddress> {
+        self.sponsor.map(SuiAddress::from)
+    }
+
+    pub fn gas_price(&self) -> u64 {
+        self.gas_price
+    }
+
+    pub fn gas_budget(&self) -> u64 {
+        self.gas_budget
+    }
+
+    pub fn ids_created(&self) -> u64 {
+        self.ids_created
+    }
+
+    /// Derive a globally unique object ID by hashing self.digest | self.ids_created
+    pub fn fresh_id(&mut self) -> ObjectID {
+        let id = ObjectID::derive_id(self.digest(), self.ids_created);
+
+        self.ids_created += 1;
+        id
     }
 
-    pub fn to_bcs_legacy_context(&self) -> Vec<u8> {
-        let move_context: MoveLegacyTxContext = self.into();
+    pub fn to_bcs_legacy_context(&self, is_native: bool) -> Vec<u8> {
```

Comment:
A small thing: 
All of the call-sites grab this flag from the protocol config first. Would it make more sense to just pass in the protocol config? It feels a bit better encapsulated